### DISCUSSION TITLE
Work more intuitively near edges of screen

### DISCUSSION THREAD

**Discussion by alexozer** [rxhanson/Charmstone-Community#36] [Ideas]
Hello, new Mac user here and this was my first paid app purchase. I'm really enjoying this app - the speed of randomly flicking between my commonly-used apps feels unparalleled on a laptop.

That being said, I think it would be really nice if things were handled a bit more consistently when the cursor is near a screen edge. If this is the case, moving the cursor into a screen edge can produce different results than if it were further away since Charmstone moves out of the way, but the cursor does not. This is even more likely to happen for the "far" apps. It would be nice to be able to do the flick gesture without needing to think about how close the cursor is to a screen edge.

I don't really know the most elegant way to do this. Maybe as the cursor slides into an edge, Charmstone slides out of the way by the opposite movement that the cursor would have taken if the edge weren't there? Or maybe if you start the gesture too close to an edge, Charmstone just auto-moves the cursor a certain distance away from the screen edge?

Thanks for reading!

**Comment by rxhanson** [comment]
Thanks for the feedback. It's been a while since I've thought about this, and I agree that this needs improvement. In my personal usage, I guess I have just always been in the habit of only activating Charmstone farther away from the screen edge. 

At a start, I think moving Charmstone's perception of the initial cursor position to where the center of the window is would be the most intuitive way to fix it. I don't think I'll move the cursor itself, as that would be an undesirable user experience for some users. I'll add this to my todo list for an upcoming release.

**Comment by alexozer** [comment]
Great, I'll look forward to seeing what you come up with! That does sound like a good start - hope it still ends up working well for the far gestures as well.
